### PR TITLE
Added detect contour

### PR DESCRIPTION
**Pull request by rmeertens**
Revamped https://github.com/paparazzi/paparazzi/pull/1785
Now it works, code looks nice, no warnings and actually compiles.

### CODE REVIEW

File: sw/airborne/modules/computer_vision/opencv_contour.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+extern float contour_distance_est;
+extern float contour_d_x, contour_d_y, contour_d_z;
+
+// values for thresholding the contour
+extern int lower_y, upper_y, lower_u, upper_u, lower_v, upper_v;
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/opencv_contour.h]: these should probably be either passed as args to find_contour or at least grouped in a struct to not pollute the whole namespace with these rather generic names...

Comment (rmeertens) [inline sw/airborne/modules/computer_vision/opencv_contour.h]: Agreed! Done.

**Comment by flixr** [comment]
IMHO same goes for the whole contour result stuff...

**Comment by rmeertens** [comment]
Done!

File: sw/airborne/modules/computer_vision/opencv_contour.h [hunk 2]
```diff
@@ -31,8 +31,12 @@
 extern "C" {
 #endif
 
-extern float contour_distance_est;
-extern float contour_d_x, contour_d_y, contour_d_z;
+struct contour_estimation{
+
+	float contour_d_x, contour_d_y, contour_d_z;
```
Comment (flixr) [inline sw/airborne/modules/computer_vision/opencv_contour.h]: please fix indentation in this file...

Comment (rmeertens) [inline sw/airborne/modules/computer_vision/opencv_contour.h]: done

**Comment by rmeertens** [comment]
Nice! Thank you very much.

**Pull request merged**